pkg/routeagent/controllers/route/driver.go

```diff
@@ -3,32 +3,57 @@ package route
 import (
 	"fmt"
 	"io/ioutil"
+	"net"
 
 	"github.com/vishvananda/netlink"
 	"k8s.io/klog"
+
+	"github.com/submariner-io/submariner/pkg/util"
 )
 
-const FlannelInterface = "flannel.1"
+func interfaceExists(iface string) bool {
+	_, err := netlink.LinkByName(iface)
+	return err == nil
+}
 
-func isFlannelCNI() error {
-	_, err := netlink.LinkByName(FlannelInterface)
-	if err != nil {
-		return fmt.Errorf("error retrieving link by name %s: %v", FlannelInterface, err)
+func discoverCNIInterface(clusterCIDR string) string {
+	cniInterfaceList := []string{
```

Comment:
May be a we can implement a more generic way... like listing the interfaces and figuring if one of them has an IP inside the pod network.

In such way we don't need to care about which interface the CNI uses.